up.ts

```diff
@@ -109,12 +136,17 @@ const start = async function () {
         stdout: 'piped',
         stderr: 'piped'
     });
+
+    const hardhatReadyEvent = 'hardhat ready';
     // NOTE: the process should keep running until we kill it
-    pipeNamedSubprocess(cyan('Hardhat'), hardhat.stdout, Deno.stdout);
+    pipeNamedSubprocess(cyan('Hardhat'), hardhat.stdout, Deno.stdout, {
+        readyEvent: hardhatReadyEvent,
```

Comment:
I'm not the biggest fan of passing this in as an option, but since a readable stream can only be consumed once I don't see another way to do it without being able to "clone" the stream.  I know how to do this easily in node.js, but after searching and trying a few things I wasn't able to do it with Deno.
Seems like this works well for now.  If we get more complicated interactions with stdout, we can revisit.

Reply:
I think with Demo, the idea is that you create buffer streams and you copy your original into the buffer. Then you can copy that buffer stream into another, etc.. Anyway, seems like more effort than required for now. Looks good to me.